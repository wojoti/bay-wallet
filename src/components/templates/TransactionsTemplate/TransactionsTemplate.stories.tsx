import {ComponentMeta, ComponentStory} from '@storybook/react-native';

import TransactionsTemplate from './TransactionsTemplate';

// More on default export: https://storybook.js.org/docs/react/writing-stories/introduction#default-export
export default {
  title: 'Templates/TransactionsTemplate',
  component: TransactionsTemplate,
} as ComponentMeta<typeof TransactionsTemplate>;

// More on component templates: https://storybook.js.org/docs/react/writing-stories/introduction#using-args
const Template: ComponentStory<typeof TransactionsTemplate> = args => (
  <TransactionsTemplate {...args} />
);

export const Example = Template.bind({});
Example.args = {
  testId: '',
};
