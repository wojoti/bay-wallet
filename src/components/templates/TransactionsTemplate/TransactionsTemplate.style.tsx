import {StyleSheet} from 'react-native';
import {secondary} from 'theme/colors';

const style = StyleSheet.create({
  wrapper: {
    backgroundColor: secondary,
    height: '100%',
  },
  spacer: {
    width: '85%',
  },
});
export default style;
