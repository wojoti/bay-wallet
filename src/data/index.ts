export const MarketData = [
  {
    id: 'f944faa6-d6fb-4a31-b099-f3a638b3ff02',
    name: 'Bitcoin',
    value: '0.02137',
    percent: '4.5',
  },
  {
    id: '3343e4a6-fe82-4e8a-8fac-b9ca0d612242',
    name: 'Bitcoin',
    value: '0.02137',
    percent: '-4.5',
  },
  {
    id: '3b3621cc-a76a-42ad-96a2-0a74e543aeac',
    name: 'Bitcoin',
    value: '0.02137',
    percent: '4.5',
  },
  {
    id: 'cfbdff7b-55de-4058-a6cc-84ebf24a913a',
    name: 'Bitcoin',
    value: '0.02137',
    percent: '-4.5',
  },
  {
    id: 'f2a5713a-cb75-4460-ae22-29a98432aa47',
    name: 'Bitcoin',
    value: '0.02137',
    percent: '4.5',
  },
  {
    id: 'a8f5e82d-948a-4974-9c26-a6b3b5288659',
    name: 'Bitcoin',
    value: '0.02137',
    percent: '-4.5',
  },
  {
    id: '324deaaf-fdc3-4e5c-b81c-2a15b50a2497',
    name: 'Bitcoin',
    value: '0.02137',
    percent: '-4.5',
  },
  {
    id: '52b72c45-f9d5-440c-ae3f-e3192c17128d',
    name: 'Bitcoin',
    value: '0.02137',
    percent: '-4.5',
  },
  {
    id: 'c8c168ea-bff9-4cd6-9bb6-4b66e1f26f42',
    name: 'Bitcoin',
    value: '0.02137',
    percent: '4.5',
  },
  {
    id: 'dc0254c6-e503-4cc3-a4db-2a57adad9c7b',
    name: 'Bitcoin',
    value: '0.02137',
    percent: '-4.5',
  },
];

export const TransactionsData = [
  {
    id: '1',
    expense: true,
    category: 'Withdraw',
    clientname: 'ATM',
    value: '-$1,000',
    date: '3d ago',
  },
  {
    id: '2',
    expense: true,
    category: 'Transfer',
    clientname: 'Jan Kowalski',
    value: '-$300',
    date: '4d ago',
  },
  {
    id: '3',
    expense: false,
    category: 'Transfer',
    clientname: 'Rudy Tabuti',
    value: '$400',
    date: '4d ago',
  },
  {
    id: '4',
    expense: false,
    category: 'Top Up',
    clientname: 'Purple Bank',
    value: '$850',
    date: '6d ago',
  },
];

export const CardsData = [
  {
    id: '1',
    name: 'Main Wallet',
    value: '$45.500,12',
    number: '444221224123',
    cardType: 'purple',
  },
  {
    id: '2',
    name: 'XYZ Wallet',
    value: '$250,2',
    number: '444221224123',
    cardType: 'green',
  },
  {
    id: '3',
    name: 'ABC Wallet',
    value: '$4.500,12',
    number: '444221224123',
    cardType: 'pink',
  },
  {
    id: '4',
    name: 'ASDF Wallet',
    value: '$450.500,72',
    number: '444221224123',
    cardType: 'purple',
  },
  {
    id: '5',
    name: 'Dark Wallet',
    value: '$21,37',
    number: '444221224123',
    cardType: 'darkPurple',
  },
];
