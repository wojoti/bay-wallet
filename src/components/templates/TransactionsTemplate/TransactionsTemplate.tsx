import {Container} from 'components/atoms';
import {
  SearchBar,
  TransactionsFilter,
  TransactionsItem,
} from 'components/molecules';
import {TransactionsData} from 'data/index';
import {PropsWithChildren} from 'react';
import {FlatList} from 'react-native-gesture-handler';
import style from './TransactionsTemplate.style';

export type TransactionsData = {
  id: string;
  expense: boolean;
  category: string;
  clientname: string;
  value: string;
  date: string;
};
export type TransactionsTemplateProps = PropsWithChildren<{
  onSearchChange: (value: string) => void;
  onSearchPress: () => void;
  searchValue: string;
  data: TransactionsData[];
  onFilterPress: (id: number) => void;
  selectedId: number;
  testId?: string;
}>;

const TransactionsTemplate = ({
  onSearchChange,
  onSearchPress,
  searchValue,
  onFilterPress,
  selectedId,
  data,
  testId,
}: TransactionsTemplateProps) => {
  return (
    <Container customStyle={style.wrapper} testId={testId} align="center">
      <Container customStyle={style.spacer}>
        <SearchBar
          onChange={onSearchChange}
          onPress={onSearchPress}
          value={searchValue}
          testId="transactionstemplate-searchbar"
        />
      </Container>
      <Container customStyle={style.spacer}>
        <TransactionsFilter
          onFilterPress={onFilterPress}
          selectedId={selectedId}
        />
      </Container>
      <Container>
        <FlatList
          data={data}
          renderItem={({item}) => (
            <TransactionsItem
              testId={'transactionslist-transactionsitem-' + item.id}
              expense={item.expense}
              category={item.category}
              clientname={item.clientname}
              value={item.value}
              date={item.date}
            />
          )}
          keyExtractor={item => item.id}
          testID="transactionstemplate-flatlist"
        />
      </Container>
    </Container>
  );
};

export default TransactionsTemplate;
