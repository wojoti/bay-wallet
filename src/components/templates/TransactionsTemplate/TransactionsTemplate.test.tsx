import '@testing-library/jest-native/extend-expect';
import {fireEvent, render, screen, within} from '@testing-library/react-native';
import {TransactionsData} from 'data/index';
import TestRenderer from 'react-test-renderer';
import TransactionsTemplate, {
  TransactionsTemplateProps,
} from './TransactionsTemplate';

let value = '';
const mockOnSearchChange = jest.fn(inputValue => (value = inputValue));
let id = 0;
const mockOnFilterPress = jest.fn(fnId => (id = fnId));
const mockOnSearchPress = jest.fn();
const props: TransactionsTemplateProps = {
  testId: 'test-transactionstemplate-id',
  onSearchChange: mockOnSearchChange,
  onSearchPress: mockOnSearchPress,
  onFilterPress: mockOnFilterPress,
  selectedId: id,
  searchValue: value,
  data: TransactionsData,
};

test('should match snapshot', () => {
  const tree = TestRenderer.create(
    <TransactionsTemplate {...props} />,
  ).toJSON();
  expect(tree).toMatchSnapshot();
});

test('should render transactionstemplate', () => {
  render(<TransactionsTemplate {...props} />);
  const transactionstemplateElement = screen.getByTestId(
    'test-transactionstemplate-id',
  );
  expect(transactionstemplateElement).toBeOnTheScreen();
});

test('should render SearchBar', () => {
  render(<TransactionsTemplate {...props} />);
  const transactionstemplateElement = screen.getByTestId(
    'test-transactionstemplate-id',
  );
  expect(transactionstemplateElement).toBeOnTheScreen();
  const SearchBarElement = within(transactionstemplateElement).getByTestId(
    'transactionstemplate-searchbar',
  );
  expect(SearchBarElement).toBeOnTheScreen();
});

test('should render SearchBar - input', () => {
  render(<TransactionsTemplate {...props} />);
  const transactionstemplateElement = screen.getByTestId(
    'test-transactionstemplate-id',
  );
  expect(transactionstemplateElement).toBeOnTheScreen();
  const SearchBarElement = within(transactionstemplateElement).getByTestId(
    'transactionstemplate-searchbar',
  );
  expect(SearchBarElement).toBeOnTheScreen();
  const inputElement = within(SearchBarElement).getByTestId('searchbar-input');
  expect(inputElement).toBeOnTheScreen();
});

test('should render SearchBar - input - handle onChange', () => {
  render(<TransactionsTemplate {...props} />);
  const transactionstemplateElement = screen.getByTestId(
    'test-transactionstemplate-id',
  );
  expect(transactionstemplateElement).toBeOnTheScreen();
  const SearchBarElement = within(transactionstemplateElement).getByTestId(
    'transactionstemplate-searchbar',
  );
  expect(SearchBarElement).toBeOnTheScreen();
  const inputElement = within(SearchBarElement).getByTestId('searchbar-input');
  expect(inputElement).toBeOnTheScreen();
  fireEvent.changeText(inputElement, 'test1234');
  expect(mockOnSearchChange).toHaveBeenCalledTimes(1);
  expect(value).toBe('test1234');
});

test('should render SearchBar - button', () => {
  render(<TransactionsTemplate {...props} />);
  const transactionstemplateElement = screen.getByTestId(
    'test-transactionstemplate-id',
  );
  expect(transactionstemplateElement).toBeOnTheScreen();
  const SearchBarElement = within(transactionstemplateElement).getByTestId(
    'transactionstemplate-searchbar',
  );
  expect(SearchBarElement).toBeOnTheScreen();
  const buttonElement =
    within(SearchBarElement).getByTestId('searchbar-button');
  expect(buttonElement).toBeOnTheScreen();
});

test('should render SearchBar - button - handle onPress', () => {
  render(<TransactionsTemplate {...props} />);
  const transactionstemplateElement = screen.getByTestId(
    'test-transactionstemplate-id',
  );
  expect(transactionstemplateElement).toBeOnTheScreen();
  const SearchBarElement = within(transactionstemplateElement).getByTestId(
    'transactionstemplate-searchbar',
  );
  expect(SearchBarElement).toBeOnTheScreen();
  const buttonElement =
    within(SearchBarElement).getByTestId('searchbar-button');
  expect(buttonElement).toBeOnTheScreen();
  expect(mockOnSearchPress).toHaveBeenCalledTimes(0);
  fireEvent.press(buttonElement);
  expect(mockOnSearchPress).toHaveBeenCalledTimes(1);
});
