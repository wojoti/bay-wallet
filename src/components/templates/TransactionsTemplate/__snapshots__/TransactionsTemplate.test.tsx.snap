// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match snapshot 1`] = `
<View
  style={
    [
      {
        "alignItems": "center",
        "flex": undefined,
        "flexDirection": "column",
        "justifyContent": "flex-start",
        "width": "100%",
      },
      {
        "backgroundColor": "white",
        "height": "100%",
      },
    ]
  }
  testID="test-transactionstemplate-id"
>
  <View
    style={
      [
        {
          "alignItems": "stretch",
          "flex": undefined,
          "flexDirection": "column",
          "justifyContent": "flex-start",
          "width": "100%",
        },
        {
          "width": "85%",
        },
      ]
    }
  >
    <View
      style={
        [
          {
            "alignItems": "center",
            "flex": undefined,
            "flexDirection": "row",
            "justifyContent": "space-between",
            "width": "100%",
          },
          {
            "backgroundColor": "#F5F5F5",
            "borderRadius": 8,
          },
        ]
      }
      testID="transactionstemplate-searchbar"
    >
      <TextInput
        accessibilityHint="Type to filter transactions"
        accessibilityLabel="Search bar"
        autoComplete="off"
        autoCorrect={false}
        onChangeText={[MockFunction]}
        placeholder="Search here"
        secureTextEntry={false}
        style={
          [
            {
              "borderColor": "#363062",
              "borderRadius": 8,
              "borderWidth": 1,
              "fontFamily": "Poppins-Light",
              "fontSize": 16,
              "fontWeight": "400",
              "height": 65,
              "paddingLeft": 30,
              "width": "100%",
            },
            {
              "borderWidth": 0,
              "width": "80%",
            },
          ]
        }
        testID="searchbar-input"
        value=""
      />
      <View
        style={
          [
            {
              "alignItems": "stretch",
              "flex": undefined,
              "flexDirection": "column",
              "justifyContent": "flex-start",
              "width": "100%",
            },
            {
              "paddingRight": 30,
              "width": "auto",
            },
          ]
        }
      >
        <View
          accessibilityRole="button"
          accessibilityState={
            {
              "busy": undefined,
              "checked": undefined,
              "disabled": false,
              "expanded": undefined,
              "selected": undefined,
            }
          }
          accessibilityValue={
            {
              "max": undefined,
              "min": undefined,
              "now": undefined,
              "text": undefined,
            }
          }
          accessible={true}
          collapsable={false}
          focusable={true}
          onBlur={[Function]}
          onClick={[Function]}
          onFocus={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            [
              {
                "alignItems": "center",
                "backgroundColor": "transparent",
                "borderStyle": "solid",
                "justifyContent": "center",
                "opacity": 1,
              },
              false,
              undefined,
            ]
          }
          testID="searchbar-button"
        >
          <SvgMock
            height="24"
            width={24}
          />
        </View>
      </View>
    </View>
  </View>
  <View
    style={
      [
        {
          "alignItems": "stretch",
          "flex": undefined,
          "flexDirection": "column",
          "justifyContent": "flex-start",
          "width": "100%",
        },
        {
          "width": "85%",
        },
      ]
    }
  >
    <View
      style={
        [
          {
            "alignItems": "center",
            "flex": undefined,
            "flexDirection": "row",
            "justifyContent": "flex-start",
            "width": "100%",
          },
          undefined,
        ]
      }
    >
      <View
        accessibilityRole="button"
        accessibilityState={
          {
            "busy": undefined,
            "checked": undefined,
            "disabled": false,
            "expanded": undefined,
            "selected": undefined,
          }
        }
        accessibilityValue={
          {
            "max": undefined,
            "min": undefined,
            "now": undefined,
            "text": undefined,
          }
        }
        accessible={true}
        collapsable={false}
        focusable={true}
        onBlur={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          [
            {
              "alignItems": "center",
              "borderRadius": 8,
              "borderStyle": "solid",
              "borderWidth": 1,
              "height": 65,
              "justifyContent": "center",
              "opacity": 1,
            },
            false,
            {
              "borderWidth": 0,
              "marginRight": 20,
            },
          ]
        }
        testID="transactionsfilter-button-all"
      >
        <Text
          style={
            [
              {
                "fontFamily": "Poppins-Bold",
                "fontSize": 18,
                "textAlign": "center",
              },
              {
                "color": "#BDBDBD",
              },
            ]
          }
          testID="button-text"
        >
          All
        </Text>
      </View>
      <View
        accessibilityRole="button"
        accessibilityState={
          {
            "busy": undefined,
            "checked": undefined,
            "disabled": false,
            "expanded": undefined,
            "selected": undefined,
          }
        }
        accessibilityValue={
          {
            "max": undefined,
            "min": undefined,
            "now": undefined,
            "text": undefined,
          }
        }
        accessible={true}
        collapsable={false}
        focusable={true}
        onBlur={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          [
            {
              "alignItems": "center",
              "borderRadius": 8,
              "borderStyle": "solid",
              "borderWidth": 1,
              "height": 65,
              "justifyContent": "center",
              "opacity": 1,
            },
            false,
            {
              "borderWidth": 0,
              "marginRight": 20,
            },
          ]
        }
        testID="transactionsfilter-button-expense"
      >
        <Text
          style={
            [
              {
                "fontFamily": "Poppins-Bold",
                "fontSize": 18,
                "textAlign": "center",
              },
              {
                "color": "#BDBDBD",
              },
            ]
          }
          testID="button-text"
        >
          Expense
        </Text>
      </View>
      <View
        accessibilityRole="button"
        accessibilityState={
          {
            "busy": undefined,
            "checked": undefined,
            "disabled": false,
            "expanded": undefined,
            "selected": undefined,
          }
        }
        accessibilityValue={
          {
            "max": undefined,
            "min": undefined,
            "now": undefined,
            "text": undefined,
          }
        }
        accessible={true}
        collapsable={false}
        focusable={true}
        onBlur={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          [
            {
              "alignItems": "center",
              "borderRadius": 8,
              "borderStyle": "solid",
              "borderWidth": 1,
              "height": 65,
              "justifyContent": "center",
              "opacity": 1,
            },
            false,
            {
              "borderWidth": 0,
              "marginRight": 20,
            },
          ]
        }
        testID="transactionsfilter-button-income"
      >
        <Text
          style={
            [
              {
                "fontFamily": "Poppins-Bold",
                "fontSize": 18,
                "textAlign": "center",
              },
              {
                "color": "#BDBDBD",
              },
            ]
          }
          testID="button-text"
        >
          Income
        </Text>
      </View>
    </View>
  </View>
  <View
    style={
      [
        {
          "alignItems": "stretch",
          "flex": undefined,
          "flexDirection": "column",
          "justifyContent": "flex-start",
          "width": "100%",
        },
        undefined,
      ]
    }
  >
    <RCTScrollView
      collapsable={false}
      data={
        [
          {
            "category": "Withdraw",
            "clientname": "ATM",
            "date": "3d ago",
            "expense": true,
            "id": "1",
            "value": "-$1,000",
          },
          {
            "category": "Transfer",
            "clientname": "Jan Kowalski",
            "date": "4d ago",
            "expense": true,
            "id": "2",
            "value": "-$300",
          },
          {
            "category": "Transfer",
            "clientname": "Rudy Tabuti",
            "date": "4d ago",
            "expense": false,
            "id": "3",
            "value": "$400",
          },
          {
            "category": "Top Up",
            "clientname": "Purple Bank",
            "date": "6d ago",
            "expense": false,
            "id": "4",
            "value": "$850",
          },
        ]
      }
      getItem={[Function]}
      getItemCount={[Function]}
      handlerTag={1}
      handlerType="NativeViewGestureHandler"
      keyExtractor={[Function]}
      onContentSizeChange={[Function]}
      onGestureHandlerEvent={[Function]}
      onGestureHandlerStateChange={[Function]}
      onLayout={[Function]}
      onMomentumScrollBegin={[Function]}
      onMomentumScrollEnd={[Function]}
      onScroll={[Function]}
      onScrollBeginDrag={[Function]}
      onScrollEndDrag={[Function]}
      removeClippedSubviews={false}
      renderItem={[Function]}
      renderScrollComponent={[Function]}
      scrollEventThrottle={50}
      stickyHeaderIndices={[]}
      testID="transactionstemplate-flatlist"
      viewabilityConfigCallbackPairs={[]}
      waitFor={
        [
          {
            "current": null,
          },
          {
            "current": null,
          },
        ]
      }
    >
      <View>
        <View
          onFocusCapture={[Function]}
          onLayout={[Function]}
          style={null}
        >
          <View
            style={
              [
                {
                  "alignItems": "center",
                  "flex": undefined,
                  "flexDirection": "row",
                  "justifyContent": "space-between",
                  "width": "100%",
                },
                {
                  "marginVertical": 17,
                  "paddingHorizontal": 30,
                },
              ]
            }
            testID="transactionslist-transactionsitem-1"
          >
            <View
              style={
                [
                  {
                    "alignItems": "center",
                    "flex": 4,
                    "flexDirection": "row",
                    "justifyContent": "flex-start",
                    "width": "100%",
                  },
                  undefined,
                ]
              }
            >
              <View
                style={
                  [
                    {
                      "alignItems": "stretch",
                      "flex": undefined,
                      "flexDirection": "column",
                      "justifyContent": "flex-start",
                      "width": "100%",
                    },
                    {
                      "alignItems": "center",
                      "backgroundColor": "#346EC5",
                      "borderRadius": 8,
                      "height": 52,
                      "justifyContent": "center",
                      "marginRight": 15,
                      "width": 52,
                    },
                  ]
                }
              >
                <SvgMock
                  height="21"
                  testID="transactionsitem-img"
                  width="100%"
                />
              </View>
              <View
                style={
                  [
                    {
                      "alignItems": "flex-start",
                      "flex": undefined,
                      "flexDirection": "column",
                      "justifyContent": "center",
                      "width": "100%",
                    },
                    undefined,
                  ]
                }
              >
                <Text
                  style={
                    {
                      "color": "#3D454B",
                      "fontFamily": "Poppins-Light",
                      "fontSize": 14,
                      "fontWeight": "400",
                      "textAlign": "left",
                    }
                  }
                  testID="transactionsitem-category"
                >
                  Withdraw
                </Text>
                <Text
                  style={
                    {
                      "color": "black",
                      "fontFamily": "Poppins-Bold",
                      "fontSize": 16,
                      "textAlign": "left",
                    }
                  }
                  testID="transactionsitem-clientname"
                >
                  ATM
                </Text>
              </View>
            </View>
            <View
              style={
                [
                  {
                    "alignItems": "flex-end",
                    "flex": 1,
                    "flexDirection": "column",
                    "justifyContent": "center",
                    "width": "100%",
                  },
                  undefined,
                ]
              }
            >
              <Text
                style={
                  {
                    "color": "#E52020",
                    "fontFamily": "Poppins-Bold",
                    "fontSize": 16,
                    "textAlign": "left",
                  }
                }
                testID="transactionsitem-value"
              >
                -$1,000
              </Text>
              <Text
                style={
                  {
                    "color": "#6B6B6B",
                    "fontFamily": "Poppins-Light",
                    "fontSize": 14,
                    "fontWeight": "400",
                    "textAlign": "left",
                  }
                }
                testID="transactionsitem-date"
              >
                3d ago
              </Text>
            </View>
          </View>
        </View>
        <View
          onFocusCapture={[Function]}
          onLayout={[Function]}
          style={null}
        >
          <View
            style={
              [
                {
                  "alignItems": "center",
                  "flex": undefined,
                  "flexDirection": "row",
                  "justifyContent": "space-between",
                  "width": "100%",
                },
                {
                  "marginVertical": 17,
                  "paddingHorizontal": 30,
                },
              ]
            }
            testID="transactionslist-transactionsitem-2"
          >
            <View
              style={
                [
                  {
                    "alignItems": "center",
                    "flex": 4,
                    "flexDirection": "row",
                    "justifyContent": "flex-start",
                    "width": "100%",
                  },
                  undefined,
                ]
              }
            >
              <View
                style={
                  [
                    {
                      "alignItems": "stretch",
                      "flex": undefined,
                      "flexDirection": "column",
                      "justifyContent": "flex-start",
                      "width": "100%",
                    },
                    {
                      "alignItems": "center",
                      "backgroundColor": "#346EC5",
                      "borderRadius": 8,
                      "height": 52,
                      "justifyContent": "center",
                      "marginRight": 15,
                      "width": 52,
                    },
                  ]
                }
              >
                <SvgMock
                  height="21"
                  testID="transactionsitem-img"
                  width="100%"
                />
              </View>
              <View
                style={
                  [
                    {
                      "alignItems": "flex-start",
                      "flex": undefined,
                      "flexDirection": "column",
                      "justifyContent": "center",
                      "width": "100%",
                    },
                    undefined,
                  ]
                }
              >
                <Text
                  style={
                    {
                      "color": "#3D454B",
                      "fontFamily": "Poppins-Light",
                      "fontSize": 14,
                      "fontWeight": "400",
                      "textAlign": "left",
                    }
                  }
                  testID="transactionsitem-category"
                >
                  Transfer
                </Text>
                <Text
                  style={
                    {
                      "color": "black",
                      "fontFamily": "Poppins-Bold",
                      "fontSize": 16,
                      "textAlign": "left",
                    }
                  }
                  testID="transactionsitem-clientname"
                >
                  Jan Kowalski
                </Text>
              </View>
            </View>
            <View
              style={
                [
                  {
                    "alignItems": "flex-end",
                    "flex": 1,
                    "flexDirection": "column",
                    "justifyContent": "center",
                    "width": "100%",
                  },
                  undefined,
                ]
              }
            >
              <Text
                style={
                  {
                    "color": "#E52020",
                    "fontFamily": "Poppins-Bold",
                    "fontSize": 16,
                    "textAlign": "left",
                  }
                }
                testID="transactionsitem-value"
              >
                -$300
              </Text>
              <Text
                style={
                  {
                    "color": "#6B6B6B",
                    "fontFamily": "Poppins-Light",
                    "fontSize": 14,
                    "fontWeight": "400",
                    "textAlign": "left",
                  }
                }
                testID="transactionsitem-date"
              >
                4d ago
              </Text>
            </View>
          </View>
        </View>
        <View
          onFocusCapture={[Function]}
          onLayout={[Function]}
          style={null}
        >
          <View
            style={
              [
                {
                  "alignItems": "center",
                  "flex": undefined,
                  "flexDirection": "row",
                  "justifyContent": "space-between",
                  "width": "100%",
                },
                {
                  "marginVertical": 17,
                  "paddingHorizontal": 30,
                },
              ]
            }
            testID="transactionslist-transactionsitem-3"
          >
            <View
              style={
                [
                  {
                    "alignItems": "center",
                    "flex": 4,
                    "flexDirection": "row",
                    "justifyContent": "flex-start",
                    "width": "100%",
                  },
                  undefined,
                ]
              }
            >
              <View
                style={
                  [
                    {
                      "alignItems": "stretch",
                      "flex": undefined,
                      "flexDirection": "column",
                      "justifyContent": "flex-start",
                      "width": "100%",
                    },
                    {
                      "alignItems": "center",
                      "backgroundColor": "#346EC5",
                      "borderRadius": 8,
                      "height": 52,
                      "justifyContent": "center",
                      "marginRight": 15,
                      "width": 52,
                    },
                  ]
                }
              >
                <SvgMock
                  height="21"
                  testID="transactionsitem-img"
                  width="100%"
                />
              </View>
              <View
                style={
                  [
                    {
                      "alignItems": "flex-start",
                      "flex": undefined,
                      "flexDirection": "column",
                      "justifyContent": "center",
                      "width": "100%",
                    },
                    undefined,
                  ]
                }
              >
                <Text
                  style={
                    {
                      "color": "#3D454B",
                      "fontFamily": "Poppins-Light",
                      "fontSize": 14,
                      "fontWeight": "400",
                      "textAlign": "left",
                    }
                  }
                  testID="transactionsitem-category"
                >
                  Transfer
                </Text>
                <Text
                  style={
                    {
                      "color": "black",
                      "fontFamily": "Poppins-Bold",
                      "fontSize": 16,
                      "textAlign": "left",
                    }
                  }
                  testID="transactionsitem-clientname"
                >
                  Rudy Tabuti
                </Text>
              </View>
            </View>
            <View
              style={
                [
                  {
                    "alignItems": "flex-end",
                    "flex": 1,
                    "flexDirection": "column",
                    "justifyContent": "center",
                    "width": "100%",
                  },
                  undefined,
                ]
              }
            >
              <Text
                style={
                  {
                    "color": "#61C277",
                    "fontFamily": "Poppins-Bold",
                    "fontSize": 16,
                    "textAlign": "left",
                  }
                }
                testID="transactionsitem-value"
              >
                $400
              </Text>
              <Text
                style={
                  {
                    "color": "#6B6B6B",
                    "fontFamily": "Poppins-Light",
                    "fontSize": 14,
                    "fontWeight": "400",
                    "textAlign": "left",
                  }
                }
                testID="transactionsitem-date"
              >
                4d ago
              </Text>
            </View>
          </View>
        </View>
        <View
          onFocusCapture={[Function]}
          onLayout={[Function]}
          style={null}
        >
          <View
            style={
              [
                {
                  "alignItems": "center",
                  "flex": undefined,
                  "flexDirection": "row",
                  "justifyContent": "space-between",
                  "width": "100%",
                },
                {
                  "marginVertical": 17,
                  "paddingHorizontal": 30,
                },
              ]
            }
            testID="transactionslist-transactionsitem-4"
          >
            <View
              style={
                [
                  {
                    "alignItems": "center",
                    "flex": 4,
                    "flexDirection": "row",
                    "justifyContent": "flex-start",
                    "width": "100%",
                  },
                  undefined,
                ]
              }
            >
              <View
                style={
                  [
                    {
                      "alignItems": "stretch",
                      "flex": undefined,
                      "flexDirection": "column",
                      "justifyContent": "flex-start",
                      "width": "100%",
                    },
                    {
                      "alignItems": "center",
                      "backgroundColor": "#346EC5",
                      "borderRadius": 8,
                      "height": 52,
                      "justifyContent": "center",
                      "marginRight": 15,
                      "width": 52,
                    },
                  ]
                }
              >
                <SvgMock
                  height="21"
                  testID="transactionsitem-img"
                  width="100%"
                />
              </View>
              <View
                style={
                  [
                    {
                      "alignItems": "flex-start",
                      "flex": undefined,
                      "flexDirection": "column",
                      "justifyContent": "center",
                      "width": "100%",
                    },
                    undefined,
                  ]
                }
              >
                <Text
                  style={
                    {
                      "color": "#3D454B",
                      "fontFamily": "Poppins-Light",
                      "fontSize": 14,
                      "fontWeight": "400",
                      "textAlign": "left",
                    }
                  }
                  testID="transactionsitem-category"
                >
                  Top Up
                </Text>
                <Text
                  style={
                    {
                      "color": "black",
                      "fontFamily": "Poppins-Bold",
                      "fontSize": 16,
                      "textAlign": "left",
                    }
                  }
                  testID="transactionsitem-clientname"
                >
                  Purple Bank
                </Text>
              </View>
            </View>
            <View
              style={
                [
                  {
                    "alignItems": "flex-end",
                    "flex": 1,
                    "flexDirection": "column",
                    "justifyContent": "center",
                    "width": "100%",
                  },
                  undefined,
                ]
              }
            >
              <Text
                style={
                  {
                    "color": "#61C277",
                    "fontFamily": "Poppins-Bold",
                    "fontSize": 16,
                    "textAlign": "left",
                  }
                }
                testID="transactionsitem-value"
              >
                $850
              </Text>
              <Text
                style={
                  {
                    "color": "#6B6B6B",
                    "fontFamily": "Poppins-Light",
                    "fontSize": 14,
                    "fontWeight": "400",
                    "textAlign": "left",
                  }
                }
                testID="transactionsitem-date"
              >
                6d ago
              </Text>
            </View>
          </View>
        </View>
      </View>
    </RCTScrollView>
  </View>
</View>
`;
